app/services/speech_to_text.py

```
@@ -0,0 +1,110 @@
+            try:
+                transcribed_text = self._transcribe_with_openai(temp_file_path)
+            finally:
+                self._cleanup_temp_file(temp_file_path)
+
+            logger.info(f"✅ 음성 변환 완료: {transcribed_text}")
+            return transcribed_text
+
+        except Exception as e:
+            error_message = self._handle_transcription_error(str(e))
+            logger.error(f"❌ 음성 변환 실패: {error_message}")
+            raise Exception(error_message)
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)